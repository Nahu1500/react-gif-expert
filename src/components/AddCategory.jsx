import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const cleanInputValue = inputValue.trim();

        if(!(cleanInputValue.length <= 1))
            onNewCategory(cleanInputValue);

        // setCategories((categories) => [inputValue, ...categories]);

        setInputValue('');
    }

    return (
        // <form onSubmit={(event) => onSubmit(event)}> FORMA LARGA
        <form onSubmit={onSubmit} aria-label='form'>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};