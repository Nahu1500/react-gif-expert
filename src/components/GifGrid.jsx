import { GifItem } from './GifItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category, onDelete }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {/* {
                // isLoading ? (<h2>Cargando...</h2>) : null
                isLoading && (<h2>Cargando...</h2>) // Más simple
            } */}


            <div className='card-grid'>
                {
                    images.map((img) => (
                        <GifItem
                            key={img.id}
                            // title={img.title}
                            // url={img.url}
                            {...img} //Me permite hacer lo mismo de arriba pero más corto, manda todos los atributos del objeto como props al componente
                        />
                    ))
                }
            </div>
            
            {
                isLoading
                    ? (<h2>Cargando...</h2>)
                    : <button className="button" onClick={() => onDelete(category)}>Eliminar</button>
            }
            {/* <button className="button" onClick={() => onDelete(category)}>Eliminar</button> */}
        </>
    )
}
