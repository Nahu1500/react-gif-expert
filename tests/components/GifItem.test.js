import {render, screen} from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {

    const title = "Hola";
    const url = "https://giphy.com/embed/hs67xo8fGYfx5KlBgV";

    test('Debe de hacer match con el Snapshot', () => {
        const {container} = render(<GifItem  title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem  title={title} url={url}/>);
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render(<GifItem  title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
    })
})