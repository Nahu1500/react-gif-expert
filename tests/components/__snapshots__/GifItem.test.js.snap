// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="https://giphy.com/embed/hs67xo8fGYfx5KlBgV"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
